pkg/registry/build_registry/build_registry.go

```diff
@@ -167,10 +167,7 @@ func generateIEString(name string, elementid string, datatype string, enterprise
 	elementID, _ := strconv.ParseUint(elementid, 10, 16)
 	enterpriseID, _ := strconv.ParseUint(enterpriseid, 10, 16)
 	dataType := entities.IENameToType(datatype)
-	length := uint16(0)
-	if entities.IsValidDataType(dataType) {
```

Comment:
This "if" check can be removed even if `InfoElementLength` is a list too right as there is an index for `invalidDataType`.

Reply:
If InfoElementLength is a list, removing the checking will cause list out-of-range error because InvalidDataType is 255 while list length is only 24.

Reply:
Sure. Misled by the last element in the list which is 0 that has index 23--I think it was intended for InvalidDataType but was not used at all.

Another alternative is to continue with a list and avoid hardcoding InvalidDataType to be 255 and instead have the next iterated value in enum dataTypes. What do you think of this option?

Reply:
If InvalidDataType is only used internally, then it is OK.
Personally I prefer using a map because it is easier to insert/delete/update types.
Maintaining the indices and values in two different structures are more error-prone in my experiences.
But this is just my opinion. Thanks.

Reply:
Thanks Shihhao. We add and access the elements in the map/list only once when registering InfoElements from registries and do not modify it at all. The rest of the code uses the length in InfoElement struct directly. In addition, when we restrict adding the elements with invalidDataType to the registry, the usage comes further down.

Either way, it is ok, but because of its minimal usage, we may go with a list instead of a map.

Reply:
My concert is more about the readability of the code.
I think keeping keys and values in the same data structure is easier to maintain.
Whoever needs to add/delete/update an entry doesn't need to count the offset in one place and change the value in another place. Does it make sense? Thanks.

Reply:
Sure from a code readability point of view, it is beneficial to have a map. Since it is not very large, it is ok from a memory overhead point of view. We can use a map.